thonny/editors.py

```diff
@@ -92,8 +92,15 @@ def update_file_type(self):
         if self._filename is None:
             self._code_view.set_file_type(None)
         else:
-            ext = self._filename.split(".")[-1].lower()
-            if ext in PYTHON_EXTENSIONS:
+            # Include backup files: *.py.bak, *.py.orig, etc
+            get_dots = len(re.findall(r'\.', self._filename))
+            ext = self._filename.split(".")[-get_dots].lower()
+
+            # Extract the 1st line to get the env
+            with open(self._filename, 'r') as shebang:
```

Comment:
You should consult the editor content, not the file, as the file may be located in another machine.

Reply:
I've fixed all the other things, but this one… Well, there's been a lot of trial'n'error. :) I don't now, but I think I'm on to it. Just can't get the output/content. If you could provide some help/tips, I would _really_ appreciate that.

So, after trying a lot of different things - printing out a lot to terminal. With `self.notebook.get_editor(self.get_filename(), True)` I can get the editor, and I can get it to print out the filename, when I open a file.

```py
e = self.notebook.get_editor(self.get_filename(), True)

# e have: _name, _filename and _code_view
e._name                 # !editor2  
e._filename             # /path/2/my/file.py
e._code_view.text       # .!codeview2.!editorcodeviewtext
```

`e._code_view.text` have `get`, so I thought just grabbing the first line would be great, but it's empty.

```py
e._code_view.text.get('1.0', '1.end')    # empty
e._code_view.get_content()               # empty
```

Ideally, one would just want to use something like: `.text.get('1.0', '1.end').find('python')`. But, I'm not _too_ familiar with tk, so I feel like I'm a bit in the dark now - trying and guessing.

Any ideas? Or maybe I'm looking at the wrong place/function(s)?